src/ParameterDistributions.jl

```diff
@@ -0,0 +1,321 @@
+                          constraints::Array{CType},
+                          name::String) where {PDType <: ParameterDistributionType,
+                                               CType <: ConstraintType} = !(dimension(parameter_distribution) ==
+len(constraints)) ? throw(DimensionMismatch("There must be one constraint per parameter")) : new{PDType,CType}(parameter_distribution, constraints, name)     
+end
+    
+
+"""
+    struct ParameterDistributions
+
+Structure to hold an array of ParameterDistribution's
+"""
+struct ParameterDistributions
+    parameter_distributions::Array{ParameterDistribution}
+end
+
+
+## Functions
+
+"""
+    get_name(pds::ParameterDistributions)
+
+Returns a list of ParameterDistribution names
+"""
+function get_name(pds::ParameterDistributions)
+    return [get_name(pd) for pd in pds.parameter_distributions]
+end
+
+function get_name(pd::ParameterDistribution)
+    return pd.name
+end
+
+"""
+    get_distributions(pds::ParameterDistributions)
+
+Returns a `Dict` of `ParameterDistribution` distributions by name, (unless sample type)
+"""
+function get_distribution(pds::ParameterDistributions)
+    return Dict{String,Any}(get_name(pd) => get_distribution(pd) for pd in pds.parameter_distributions)
+end
+
+function get_distribution(pd::ParameterDistribution)
+    return get_distribution(pd.distribution)
+end
+
+function get_distribution(d::Samples)
+    return "Contains samples only"
+end
+function get_distribution(d::Parameterized)
+    return d.distribution
+end
+
+"""
+    function sample_distribution(pds::ParameterDistributions)
+
+Draws samples from the parameter distributions
+"""
+function sample_distribution(pds::ParameterDistributions)
+    return sample_distribution(pds,1)
+end
+
+function sample_distribution(pds::ParameterDistributions, n_draws::IT) where {IT <: Integer}
+    return cat([sample_distribution(pd,n_draws) for pd in pds.parameter_distributions]...,dims=1)
+end
+
+function sample_distribution(pd::ParameterDistribution)
+    return sample_distribution(pd,1)
+end
+
+function sample_distribution(pd::ParameterDistribution, n_draws::IT) where {IT <: Integer}
+    return sample_distribution(pd.distribution,n_draws)
+end
+
+function sample_distribution(d::Samples, n_draws::IT)  where {IT <: Integer}
+    n_stored_samples = n_samples(d)
+    samples_idx = StatsBase.sample(collect(1:n_stored_samples), n_draws)
+    return d.distribution_samples[:,samples_idx]
+
+end
+
+function sample_distribution(d::Parameterized, n_draws::IT) where {IT <: Integer}
+    return rand(d.distribution, n_draws)
+end
+
+
+#apply transforms
+
+"""
+    transform_real_to_prior(pds::ParameterDistributions, x::Array{Real})
+
+Apply the transformation to map real (and possibly constrained) parameters `xarray` into the unbounded prior space
+"""
+function transform_real_to_prior(pds::ParameterDistributions, xarray::Array{FT}) where {FT <: Real}
+    #split xarray into chunks to be fed into each distribution
```

Comment:
The array `xarray` or `x` that we are transforming seems a more intuitive first argument to me. Could we change the order in all `transform_real_to_prior`?

Reply:
So I think these input arguments are consistent with the rest of the package in that they are pretending to act like a method in python where you call `F(self, x)` we have followed this by replacing it with  `F(self::object, x)` . Perhaps what I've done here is non-standard, but I guess writing it felt natural that the version of `F` we call is determined by it's first argument. I'm happy to change this in general, but perhaps we could do a whole-package sweep in a subsequent PR to retain consistency?